sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2ScanRelationPushDown.scala

```diff
@@ -209,6 +210,11 @@ object V2ScanRelationPushDown extends Rule[LogicalPlan] with PredicateHelper {
       }
   }
 
+  private def supportPartialAggPushDown(agg: Aggregation): Boolean = {
+    // We don't know the agg buffer of `GeneralAggregateFunc`, so can't do partial agg push down.
+    agg.aggregateExpressions().exists(_.isInstanceOf[GeneralAggregateFunc])
```

Comment:
emm..., This is broken semantics, Since can't push down if have `GeneralAggregateFunc`, Shall we return false if `agg.aggregateExpressions()` exists `GeneralAggregateFunc`, right?

`!agg.aggregateExpressions().exists(_.isInstanceOf[GeneralAggregateFunc])` or `agg.aggregateExpressions().forall(!_.isInstanceOf[GeneralAggregateFunc])`?

Reply:
+1 Looks like a bug. Shall we also add a unit test?

Reply:
The original code was correct. 
We probably want to add a test that has `avg` and multiple partitions to cover the case of `!supportPartialAggPushDown`  and `!supportCompletePushDown`

Reply:
If exists `GeneralAggregateFunc`, we cannot partial aggregate  push-down and still could complete aggregate  push-down. I think the code is correct and is not a bug.